src/components/project-input.ts

```diff
@@ -1,74 +1,129 @@
 //Project input class
-///<reference path="base-components.ts"/>
-///<reference path="../util/validation.ts"/>
-///<reference path="../decorators/autobind.ts"/>
-///<reference path="../state/project-state.ts"/>
-
-namespace App {
- export class ProjectInput extends Component<HTMLDivElement, HTMLFormElement> {
-    titleInputElement: HTMLInputElement;
-    descriptionInputElement: HTMLInputElement;
-    peopleInputElement: HTMLInputElement;
-    constructor() {
-      super("project-input", "app", true, "user-input");
-      this.titleInputElement = this.element.querySelector(
-        "#title"
-      )! as HTMLInputElement;
-      this.descriptionInputElement = this.element.querySelector(
-        "#description"
-      )! as HTMLInputElement;
-      this.peopleInputElement = this.element.querySelector(
-        "#people"
-      )! as HTMLInputElement;
-      this.configure();
-    }
-    configure() {
-      this.element.addEventListener("submit", this.submitHandler);
+///<
+import { Component } from "./base-components";
+import { Validatable, validate } from "../util/validation";
+import { projectState } from "../state/project-state";
+//ProjectInput:class which takes care of the user input with validation and event listener for submitting
+export class ProjectInput extends Component<HTMLDivElement, HTMLFormElement> {
+  titleInputElement: HTMLInputElement;
+  descriptionInputElement: HTMLInputElement;
+  peopleInputElement: HTMLInputElement;
+  constructor() {
+    //super(the id of the template we gonna use,where we gonna store the values(id of the host),
+    //if we want to add at start(true) else at the
+    //end,the id we want to give to the element if we want)
+    super("project-input", "app", true, "user-input");
+    this.titleInputElement = this.element.querySelector(
+      "#title"
+    )! as HTMLInputElement;
+    this.descriptionInputElement = this.element.querySelector(
+      "#description"
+    )! as HTMLInputElement;
+    this.peopleInputElement = this.element.querySelector(
+      "#people"
+    )! as HTMLInputElement;
+    this.configure();
+  }
+  configure() {
+    this.element.addEventListener("submit", this.submitHandler);
+  }
+  renderContent(): void {}
+  //on submit
+  private submitHandler=(event: Event)=> {
+    event.preventDefault();
+    const userInput = this.gatherUserInput();
+    //if user input is correct we receive the array
+    if (Array.isArray(userInput)) {
+      const [title, description, people] = userInput;
+      projectState.addProject(title, description, people);
+      this.clearUserInputs();
     }
-    renderContent(): void {}
-    @autobind
-    private submitHandler(event: Event): void {
-      event.preventDefault();
-      const userInput = this.gatherUserInput();
-      if (Array.isArray(userInput)) {
-        const [title, description, people] = userInput;
-        projectState.addProject(title, description, people);
-        this.clearUserInputs();
+  }
+  //cleans user input fields
```

Comment:
the function name is the same as the comment, it is descriptive enough to make the comment redundant